collect_app/src/main/java/org/odk/collect/android/formhierarchy/FormHierarchyFragment.java

```diff
@@ -792,13 +370,14 @@ private void refreshView(boolean isGoingUp) {
                 event = formController.stepToNextEvent(JavaRosaFormController.STEP_INTO_GROUP);
             }
 
-            recyclerView.setAdapter(new HierarchyListAdapter(elementsToDisplay, this::onElementClick));
+            formHierarchyViewModel.setElementsToDisplay(elementsToDisplay);
+            recyclerView.setAdapter(new HierarchyListAdapter(formHierarchyViewModel.getElementsToDisplay(), this::onElementClick));
```

Comment:
I think here you can directly use `elementsToDisplay`. However, if you always access the list via the ViewModel please update it below in line `formController.jumpToIndex(elementsToDisplay.get(0).getFormIndex());` as well.